anonlink/similarities/_dice_x86.py

```diff
@@ -83,43 +84,23 @@ def dice_coefficient_accelerated(
                f'{filter_bits})')
         raise NotImplementedError(msg)
     filter_bytes = filter_bits // 8
+    # Python char arrays of all filters from filters0 and filter1
+    carr0, carr1 = array('b'), array('b')
 
-    # Space for one filter. We will fill it with one filter from
-    # filters0 at a time.
-    carr0 = ffi.new('char[]', filter_bytes)
-
-    # Array of all filters from filters1.
-    carr1 = ffi.new(f"char[{filter_bytes * length_f1}]",
-                    b''.join(f.tobytes() for f in filters1))
-
-    c_popcounts = ffi.new(f"uint32_t[{length_f1}]",
-                          tuple(f.count() for f in filters1))
-
-    # easier to do all buffer allocations in Python and pass them to C,
-    # even for output-only arguments
-    c_scores = ffi.new("double[]", k)
-    c_indices = ffi.new("int[]", k)
-
-    for i, f0 in enumerate(filters0):
-        carr0[0:filter_bytes] = f0.tobytes()
-        matches = lib.match_one_against_many_dice_k_top(
-            carr0,
-            carr1,
-            c_popcounts,
-            length_f1,
-            filter_bytes,
-            k,
-            threshold,
-            c_indices,
-            c_scores)
-
-        if matches < 0:
-            raise RuntimeError('bad key length')
-
-        result_sims.extend(c_scores[0:matches])
-        result_indices0.extend(repeat(i, matches))
-        result_indices1.extend(c_indices[0:matches])
+    carr0.frombytes(b''.join(memoryview(f) for f in filters0))
+    carr1.frombytes(b''.join(memoryview(f) for f in filters1))
+
+    # Only worth popcounting in C for a large number of filters1
+    if len(filters1) < 10000:
```

Comment:
why 10000?

Reply:
I'll add a comment, but just found by benchmarking and noting at what size the throughput in cmp/s dropped